### PR TITLE
*: fixes based on ineffassign

### PR DESCRIPTION
**Pull request by gyuho**
To achieve 100% on https://goreportcard.com/report/github.com/coreos/etcd#ineffassign.

### CODE REVIEW

File: store/store_bench_test.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func benchStoreSet(b *testing.B, valueSize int, process func(interface{}) ([]byt
 		}
 	}
 
-	kvs = nil
```
Comment (heyitsanthony) [inline store/store_bench_test.go]: why does the tool complain here?

Comment (gyuho) [inline store/store_bench_test.go]: Think the tool is wrong. It assumes `kvs = nil` is a new assignment, so `kvs` needs to be used again.

Comment (heyitsanthony) [inline store/store_bench_test.go]: We should probably file an issue instead of doing what a broken tool suggests.

File: client/client_test.go [hunk 2]
```diff
@@ -462,7 +462,11 @@ func (f fakeCancelContext) Err() error                        { return fakeCance
 func (f fakeCancelContext) Value(key interface{}) interface{} { return 1 }
 
 func withTimeout(parent context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
-	return parent, func() { parent = nil }
+	return parent, func() { destroyCtx(parent) }
+}
+
+func destroyCtx(ctx context.Context) {
```
Comment (heyitsanthony) [inline client/client_test.go]: this is tool brokenness too?

**Comment by gyuho** [comment]
@heyitsanthony Just revoke those two changes. Will file an issue to that tool. Thanks!

File: etcdserver/api/v2http/client.go [hunk 3]
```diff
@@ -720,20 +720,20 @@ func trimEventPrefix(ev *store.Event, prefix string) *store.Event {
...
-		return nil
+		return
 	}
 	n.Key = strings.TrimPrefix(n.Key, prefix)
 	for _, nn := range n.Nodes {
-		nn = trimNodeExternPrefix(nn, prefix)
+		trimNodeExternPrefix(nn, prefix)
 	}
-	return n
+	return
```
Comment (heyitsanthony) [inline etcdserver/api/v2http/client.go]: delete?

**Comment by heyitsanthony** [comment]
lgtm

File: etcdserver/api/v2http/client.go (hunk 3 above)
Comment (gyuho) [inline etcdserver/api/v2http/client.go]: Fixed. Will merge in greens. Thanks.

**Pull request merged**